### PR TITLE
Add tests for package type pushing and filtering

### PR DESCRIPTION
**Pull request by joelverhagen**
Address https://github.com/NuGet/NuGetGallery/issues/7755

### CODE REVIEW

**Comment by zhhyu** [comment]
> using System;

Header of .net foundation?  

---
Refers to: src/NuGet.Services.EndToEnd/Support/TestData/E2E.DotnetTool/Program.cs:1 in a610340. [](commit_id = a610340ba68c718f68a582bca611d86ef421149d, deletion_comment = False)

**Comment by zhhyu** [comment]
> using System;

Header of .net foundation?

---
Refers to: src/NuGet.Services.EndToEnd/Support/TestData/E2E.DotnetTool/Program.cs:1 in a610340. [](commit_id = a610340ba68c718f68a582bca611d86ef421149d, deletion_comment = False)

File: src/NuGet.Services.EndToEnd/Support/TestData/E2E.DotnetTool/Program.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
```
Comment (zhhyu) [inline src/NuGet.Services.EndToEnd/Support/TestData/E2E.DotnetTool/Program.cs]: Header of .net foundation?

Comment (joelverhagen) [reply src/NuGet.Services.EndToEnd/Support/TestData/E2E.DotnetTool/Program.cs]: This is test content copied from the symbols package test data. This is not code compiled into the test DLL. It is actually just test data.

File: src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs [hunk 2]
```diff
@@ -46,6 +36,7 @@ public class PushedPackagesFixture : CommonFixture
         private IGalleryClient _galleryClient;
 
         private static readonly string SymbolsProjectTemplateFolder = Path.Combine(Environment.CurrentDirectory, "Support", "TestData", "E2E.TestPortableSymbols");
+        private static readonly string DotnetToolProjectTemplateFolder = Path.Combine(Environment.CurrentDirectory, "Support", "TestData", "E2E.DotnetTool");
```
Comment (zhhyu) [inline src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs]: Nit: E2E.TestDotnetTool

Comment (joelverhagen) [reply src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs]: Good idea! That will help users if they happen to see this package.

File: src/NuGet.Services.EndToEnd/Support/Package.cs [hunk 3]
```diff
@@ -68,6 +68,23 @@ public static Package Create(PackageCreationContext context)
                                context.Properties);
         }
 
+        public static Package Create(string path, PackageProperties packageProperties)
+        {
```
Comment (zhhyu) [inline src/NuGet.Services.EndToEnd/Support/Package.cs]: Null check for the arguments?

Comment (joelverhagen) [reply src/NuGet.Services.EndToEnd/Support/Package.cs]: I can add them if you feel strongly. This is test code so I don't find it as important...

File: src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs (hunk 2 above)
Comment (joelverhagen) [reply src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs]: Actually this the symbols package is called `E2E.SymbolsPackage` so the `E2E.DotnetTool` folder name matches the actual package name.

**Comment by joelverhagen** [comment]
Merging this is blocked on https://github.com/NuGet/Engineering/issues/2908.

**Pull request merged**